src/i2c/master.rs

```diff
@@ -31,6 +34,112 @@ pub enum Speed {
     High,
 }
 
+fn rounded_divide(numerator: u32, denominator: u32) -> u32 {
+    (numerator + denominator / 2) / denominator
+}
+
+struct SpeedRegisterSettings {
+    scl_high_clocks: Mstsclhigh,
+    scl_low_clocks: Mstscllow,
+    clock_div_multiplier: u16,
+
+    actual_duty_cycle: u8,
+    actual_freq_hz: u32,
+}
+
+impl SpeedRegisterSettings {
+    fn new(target_duty_cycle: u8, speed: Speed) -> Result<Self> {
+        const LO_VARIANTS: [Mstscllow; 8] = [
+            Mstscllow::Clocks2,
```

Comment:
is there a more idiomatic way to loop through enum variants like this? I couldn't find one but this seems a bit gnarly to me

Reply:
Yeah, I was hoping to do this without invoking unsafe code (and as far as I can tell, casting an integral type to an enum requires a core::mem::transmute, which is unsafe), but if that's the idiomatic way to do it then that's what I'll do.  Thanks!

Reply:
per our chat offline, can dodge unsafe code by duplicating the enum in a conversion function, which is about the same amount of code as this array nonsense I was trying, so did that